app/models/booking_response.rb

```diff
@@ -1,173 +1,133 @@
-require 'maybe_date'
-
 class BookingResponse
-  include NonPersistedModel
-
-  SLOTS = %w[ slot_0 slot_1 slot_2 ]
-
-  # The adult age for accepting a booking is different from the configurable
-  # adult age in Prison
+  include ActiveModel::Model
+  include ActiveModel::Validations::Callbacks
   ADULT_AGE = 18
+  attr_accessor :visit, :user
 
-  attribute :visit, Visit
+  before_validation :check_slot_available
+  before_validation :check_for_banned_visitors
+  before_validation :check_for_unlisted_visitors
 
-  attribute :selection, String, default: 'none'
-  validates :selection,
-    inclusion: { in: SLOTS + Rejection::REASONS },
-    if: :at_least_one_valid_visitor?
-
-  attribute :reference_no, String
-  validates :reference_no, presence: true, if: :bookable?
-  attribute :closed_visit, Virtus::Attribute::Boolean
+  validate :validate_visit_is_processable
+  validate :visit_or_rejection_validity
```

Comment:
This is not quite working, I've tried selecting a slot and a rejection and the visit was processed